src/libcephfs.cc

```diff
@@ -37,11 +37,34 @@
 #define DEFAULT_UMASK 002
 
 static mode_t umask_cb(void *);
-ceph::async::io_context_pool icp;
+namespace {
+// Set things up this way so we don't start up threads until mount and
+// kill them off when the last mount goes away, but are tolerant to
+// multiple mounts of overlapping duration.
+std::shared_ptr<ceph::async::io_context_pool> get_icp(CephContext* cct)
+{
+  std::weak_ptr<ceph::async::io_context_pool> icwp;
```

Comment:
Is this supposed to be static? I'm confused because it looks like this method always creates a new io_context_pool?

Reply:
Yes, thank you.